American Option Dynamic Prog.cpp

```diff
@@ -0,0 +1,148 @@
+// Calculating the price of an American Option using Dynamic Programming
+//where the Probability Matrix is derived from the Trinomial Lattice.
+
+#include <iostream>
+#include <iomanip>
+#include <cmath>
+#include <fstream>
+#include <cstdlib>
+#include <algorithm>
+#include ".../newmat10/newmat.h"
+using namespace std;
+
+float up_factor, uptick_prob, downtick_prob, notick_prob, risk_free_rate, strike_price;
```

Comment:
initialize these variables. If they are non-const, don't declare them as global variables. And for const, declare with const